packages/guides-restructured-text/src/RestructuredText/Directives/DefaultRoleDirective.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link https://phpdoc.org
+ */
+
+namespace phpDocumentor\Guides\RestructuredText\Directives;
+
+use phpDocumentor\Guides\Nodes\Node;
+use phpDocumentor\Guides\RestructuredText\Parser\Directive;
+use phpDocumentor\Guides\RestructuredText\Parser\DocumentParserContext;
+use phpDocumentor\Guides\RestructuredText\TextRoles\TextRoleFactory;
+
+/**
+ * sets the default interpreted text role, the role that is used for interpreted text without an explicit role.
+ *
+ * https://docutils.sourceforge.io/docs/ref/rst/directives.html#default-role
+ */
+class DefaultRoleDirective extends BaseDirective
+{
+    public function __construct(
+        private readonly TextRoleFactory $textRoleFactory,
+    ) {
+    }
+
+    public function getName(): string
+    {
+        return 'default-role';
+    }
+
+    public function process(
```

Comment:
`processAction` was intended for this type of directives. However I noticed that the base directive never calls this method, and it seems never used. Maybe we should consider to enable this usage, by introducting a new base class, that will provide an method like `processAction`

```
abstract class ActionDirective 
{
   process(): {
      $this->processAction();
      return null;
   }
   
    public function processAction(
        DocumentParserContext $documentParserContext,
        string $variable,
        string $data,
        array $options,
    ): void {
    }
}

Reply:
Good idea

Reply:
I introduced the abstract class and extended it